shared/runtime/gchelper_generic.c

```diff
@@ -150,6 +150,37 @@ static void gc_helper_get_regs(gc_helper_regs_t arr) {
     arr[10] = x29;
 }
 
+#elif defined(__riscv) && defined(__riscv_xlen) && (__riscv_xlen == 32)
+
+// Fallback implementation, prefer gchelper_rv32.s
```

Comment:
Maybe a comment about the arch: "Fallback implementation for RV32I, prefer ..."

Also, should the assembler file have the "i" in it as well, `gchelper_rv32i.s`?  In case one day we need to support RV32E.

Reply:
Oh, I copied the comment from the thumb section - I'll fix that.  Regarding the ISA, as far as I remember you can have ifdef blocks in GAS files (at least when built using `gcc`).  If that's a concern I can rename the file, that's not a problem at all.